lib_eio_luv/eio_luv.ml

```diff
@@ -455,6 +470,28 @@ module Low_level = struct
         t.fd <- `Closed;
         Eio.Switch.remove_hook t.release_hook;
         fd
+
+    let kind_map =
+      (* we only need 12, but 13 maps directly *)
+      let km = Array.init 13 (fun _ -> `Unknown) in
+      Array.set km 0o1  `Fifo;
+      Array.set km 0o2  `Character_special;
+      Array.set km 0o4  `Directory;
+      Array.set km 0o6  `Block_device;
+      Array.set km 0o10 `Regular_file;
+      Array.set km 0o12 `Symbolic_link;
+      Array.set km 0o14 `Socket;
+      km
+
+    let kind_of_mode (mode : int) : Eio.Fs.kind =
+      if Sys.os_type <> "Unix" then
+        `Unknown
+      else
+        try Array.get kind_map (mode lsr 12)
```

Comment:
Should this be masked with `S_IFMT`?

Reply:
Not needed but I'll add it for clarity, POSIX defines those 12 bits and we never look outside them.

Reply:
I've checked the actual posix specification nowand just the lower 3 bits are guarenteed to have those values, which means I had to add the discover infrastructure.